include/inexor/vulkan-renderer/world/cube.hpp

```diff
@@ -56,13 +60,17 @@ class Cube : public std::enable_shared_from_this<Cube> {
     };
 
 private:
-    Type m_type{Type::SOLID};
+    static constexpr Type DEFAULT_TYPE = Type::EMPTY;
+    Type m_type{DEFAULT_TYPE};
     float m_size{32};
     glm::vec3 m_position{0.0f, 0.0f, 0.0f};
 
     /// Root cube is empty.
     std::weak_ptr<Cube> m_parent{};
 
+    /// Index of this in m_parent.m_children; undefined behavior if root.
+    std::uint8_t m_index{};
```

Comment:
Can you name it `m_child_index`?

Reply:
I think `m_index` is more fitting because `m_child_index` would suggest an index of a child of this cube (and not the actual index of this cube) is meant.

Reply:
Maybe `m_index_in_parent`?